src/arch/armv8/aarch32/boot.S

```diff
@@ -52,12 +52,41 @@ _reset_handler:
     and r0, r0, r1
 
     /**
-     * TODO: Linearize cpu id according to the number of clusters and
-     * processors per cluster.
+     * Get base image load address.
      */
-
     adr r1, _el2_entry
 
+    /**
+     * Linearize cpu id according to the number of clusters and processors
+     * per cluster. We are only considering two levels of affinity.
+     * TODO: this should be done some other way. We shouldn't depend on the platform
+	 * description this early in the initialization.
+     */
+    mov r3, r0, lsr #8
+    and r3, r3, #0xff
+    ldr r4, =platform
+    mov r5, #(PLAT_ARCH_OFF+PLAT_ARCH_CLUSTERS_OFF+PLAT_CLUSTERS_CORES_NUM_OFF)
+    ldr r4, [r4, r5]
```

Comment:
At this point r4 contains a virtual address, you need it to first convert it to a virtual address or use a relative address instruction such as `adr` in line 67.

Reply:
You are right! Fixed using the macro `get_phys_addr` defined at the beginning of the file.